pennylane/devices/qubit/simulate.py

```diff
@@ -279,9 +279,33 @@ def simulate(
 
     has_mcm = any(isinstance(op, MidMeasureMP) for op in circuit.operations)
     if circuit.shots and has_mcm:
-        return simulate_one_shot_native_mcm(
-            circuit, debugger=debugger, rng=rng, prng_key=prng_key, interface=interface
+        results = []
+        aux_circ = qml.tape.QuantumScript(
+            circuit.operations,
+            circuit.measurements,
+            shots=[1],
+            trainable_params=circuit.trainable_params,
         )
+        keys = jax_random_split(prng_key, num=circuit.shots.total_shots)
+        if qml.math.get_deep_interface(circuit.data) == "jax":
+            # pylint: disable=import-outside-toplevel
+            import jax
+
+            def vv(k):
+                return simulate_one_shot_native_mcm(
+                    aux_circ, debugger=debugger, rng=rng, prng_key=k, interface=interface
+                )
+
+            results = jax.vmap(vv, in_axes=(0,))(keys)
```

Comment:
What would happen if a user is performing `vmap` higher in the stack?  Do nested vmap's work?

Reply:
Yes, you can do something like
```
A = np.random.rand(10, 10, 10)
B = np.random.rand(10, 10, 10)
f = jax.vmap(lambda x, y: jnp.vdot(x, y), (0, 1))
C = jax.vmap(lambda x, y: f(x, y), (0, 1))(A, B)
```